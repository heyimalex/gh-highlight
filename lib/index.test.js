const highlight = require('./index');
const { scopeNameFromLang } = highlight;

test('scopeNameFromLang', () => {
  expect(scopeNameFromLang('js')).toBe('source.js');
  expect(scopeNameFromLang('.js')).toBe('source.js');
  expect(scopeNameFromLang('javascript')).toBe('source.js');
  expect(scopeNameFromLang('JaVaScRiPt')).toBe('source.js');

  expect(scopeNameFromLang('')).toBe(undefined);
  expect(scopeNameFromLang('')).toBe(undefined);
  expect(scopeNameFromLang('  ')).toBe(undefined);
  expect(scopeNameFromLang('--unknown--')).toBe(undefined);
});

test('highlight snapshot', () => {
  expect(highlight.sync('var hello = "world";')).toMatchSnapshot();
  expect(highlight.sync('var hello = "world";', 'js')).toMatchSnapshot();
});
