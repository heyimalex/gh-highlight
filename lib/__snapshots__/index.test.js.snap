exports[`test highlight snapshot 1`] = `"<pre class=\"editor editor-colors\"><div class=\"line\"><span class=\"text plain null-grammar\"><span>var&nbsp;hello&nbsp;=&nbsp;&quot;world&quot;;</span></span></div></pre>"`;

exports[`test highlight snapshot 2`] = `"<pre class=\"editor editor-colors\"><div class=\"line\"><span class=\"source js\"><span class=\"storage type var js\"><span>var</span></span><span>&nbsp;hello&nbsp;</span><span class=\"keyword operator assignment js\"><span>=</span></span><span>&nbsp;</span><span class=\"string quoted double js\"><span class=\"punctuation definition string begin js\"><span>&quot;</span></span><span>world</span><span class=\"punctuation definition string end js\"><span>&quot;</span></span></span><span class=\"punctuation terminator statement js\"><span>;</span></span></span></div></pre>"`;
